cornichon-core/src/main/scala/com/github/agourlay/cornichon/dsl/DataTable.scala

```diff
@@ -55,20 +53,27 @@ class DataTableParser(val input: ParserInput) extends Parser with StringHeaderPa
 case class DataTable(headers: Headers, rows: Seq[Row]) {
   require(rows.forall(_.fields.size == headers.fields.size), "Datatable is malformed, all rows must have the same number of elements")
 
-  def asMap: Map[String, Seq[Json]] =
+  private def asMap: Map[String, Seq[Option[String]]] =
     headers.fields.zipWithIndex.map {
       case (header, index) ⇒
-        header → rows.map(r ⇒ r.fields(index))
+        header → rows.map(r ⇒ liftEmptyString(r.fields(index)))
     }.groupBy(_._1).map { case (k, v) ⇒ (k, v.flatMap(_._2)) }
 
+  private def liftEmptyString(s: String): Option[String] =
+    if (s.trim.nonEmpty) Some(s)
+    else None
+
   def objectList: List[JsonObject] = {
-    val tmp = for (i ← rows.indices) yield asMap.map { case (k, v) ⇒ k → v(i) }
+    val tmp = for (i ← rows.indices) yield asMap.flatMap { case (k, v) ⇒ v(i) map (sv ⇒ k → parseStringUnsafe(sv)) }
```

Comment:
I really like the change in terms of feature, plus not parsing the JSON straight in the PB parsers makes things clearer 👍 

Do you think you could you the safe version `parseString` here so that exceptions are not thrown anymore when parsing datatable?